folium/plugins/semicircle.py

```diff
@@ -0,0 +1,108 @@
+    start_angle: int, default 0
+        Heading of the start angle value between 0 and 360 degrees
+    stop_angle: int, default 0
+        Heading of the stop angle value between 0 and 360 degrees.
+    fill_color: string, default '#3388ff' (blue)
+        Fill color
+    fill_opacity: float 0-1, default 0.5
+        Fill opacity
+    color: string, default '#3388ff' (blue)
+        Line color
+    opacity: float 0-1, default 1
+        Line opacity
+    **kwargs:
+        Passed to Marker so a popup and tooltip can be provided.
+
+    """
+    _template = Template(u"""
+            {% macro script(this, kwargs) %}
+                if ({{this.direction}} || {{this.arc}}) {
+                    var {{this.get_name()}} = L.semiCircle(
+                        [{{this.location[0]}},{{this.location[1]}}],
+                        {radius:{{this.radius}},
+                        fill: {{this.fill}},
+                        fillColor:'{{this.fill_color}}',
+                        fillOpacity: {{this.fill_opacity}},
+                        color: '{{this.color}}',
+                        opacity: {{this.opacity}}
+                        }).setDirection({{this.direction}},{{this.arc}})
+                        .addTo({{this._parent.get_name()}});
+                } else if ({{this.start_angle}} || {{this.stop_angle}}) {
+                    var {{this.get_name()}} = L.semiCircle(
+                        [{{this.location[0]}},{{this.location[1]}}],
+                        {radius:{{this.radius}},
+                        fill: {{this.fill}},
+                        fillColor:'{{this.fill_color}}',
+                        fillOpacity: {{this.fill_opacity}},
+                        color: '{{this.color}}',
+                        opacity: {{this.opacity}},
+                        startAngle:{{this.start_angle}},
+                        stopAngle:{{this.stop_angle}}
+                        })
+                        .addTo({{this._parent.get_name()}});
+                }
+            {% endmacro %}
+            """)
+
+    def __init__(self, location, radius=0, direction=None, arc=None, start_angle=None, stop_angle=None, fill=True,
+                 fill_color='#3388ff', fill_opacity=0.5, color='#3388ff', opacity=1, **kwargs):
+        super(SemiCircle, self).__init__(_validate_location(location), **kwargs)
+        self._name = 'SemiCircle'
+        self.radius = radius
+        self.direction = json.dumps(direction)
+        self.arc = json.dumps(arc)
+        self.start_angle = json.dumps(start_angle)
+        self.stop_angle = json.dumps(stop_angle)
+        if fill:
+            self.fill = 'true'
+        else:
+            self.fill = 'false'
+        self.fill_color = fill_color
+        self.fill_opacity = fill_opacity
+        self.color = color
+        self.opacity = opacity
+        self.kwargs = json.dumps(kwargs)
+
+        if not (direction and arc) and not (start_angle and stop_angle):
```

Comment:
Suggestion: also raise an error if both direction and arc *and* start_angle and stop_angle are provided. If a users doesn't know that in that case the start_angle and stop_angle are ignored it may confuse them.